zgw/objecten-api/src/main/kotlin/com/ritense/objectenapi/listener/ZaakObjectListener.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.objectenapi.listener
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.JsonNodeType
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.ritense.objectenapi.ObjectenApiPlugin
+import com.ritense.objectenapi.client.ObjectRecord
+import com.ritense.objectenapi.client.ObjectRequest
+import com.ritense.objectenapi.service.ZaakObjectConstants
+import com.ritense.objectenapi.service.ZaakObjectService
+import com.ritense.plugin.service.PluginService
+import com.ritense.valtimo.contract.event.ExternalDataSubmittedEvent
+import org.springframework.context.event.EventListener
+import java.net.URI
+import java.time.LocalDate
+import java.util.InvalidPropertiesFormatException
+import java.util.UUID
+
+class ZaakObjectListener(
+    val pluginService: PluginService,
+    val zaakObjectService: ZaakObjectService,
+) {
+    @EventListener(ExternalDataSubmittedEvent::class)
+    fun handle(event: ExternalDataSubmittedEvent) {
+        event.data[ZaakObjectConstants.ZAAKOBJECT_PREFIX]?.let { it ->
+            it.entries.map {
+                RequestedField(
+                    it.key, it.value
+                )
+            }.groupBy {
+                it.objectType
+            }.forEach { objectTypeGroup ->
+                val zaakObject = zaakObjectService.getZaakObjectOfTypeByName(event.documentId, objectTypeGroup.key)
+                objectTypeGroup.value.forEach {
+                    // For each requestedField update the value in the zaakObject record data
+                    val startPath = it.path.substring(1)
+                    findAndReplaceJsonPath(zaakObject.record.data!!, startPath, extractValue(it.value as JsonNode)!!.toString())
```

Comment:
Is the value to update always a string?

Reply:
It's even more complicated. The code assumes it.value is a JsonNode, but it's actually Any. And you're right: if you would update a number field, it will become a string after the update. I'll see if I can fix this.